lib/eth/abi/type.rb

```diff
@@ -98,15 +109,13 @@ def ==(another_type)
       def size
         s = nil
         if dimensions.empty?
-          unless ["string", "bytes"].include?(base_type) and sub_type.empty?
+          if !(["string", "bytes", "tuple"].include?(base_type) and sub_type.empty?)
```

Comment:
nit: any reason you changed unless to !if?

Reply:
1. https://stackoverflow.com/a/19216609
2. https://stackoverflow.com/questions/41196492/ruby-unless-bug-with-multiple-conditions
3. https://signalvnoise.com/posts/2699-making-sense-with-rubys-unless
4. The `and` keyword having lower operator precedence than `&&` definitely didn't help with the readability

Some good rules of thumb:
1) Prefer `if` over `unless` 90% of the time. 
2) Prefer `if` over `unless` 100% of the time if there are multiple conditions.
3) Never use `and` or `or` or `not`.